File: Factorial/Factorial.java
import java.math.BigInteger;

public class Factorial {
    public static void main(String[] args) {
        long tInit = System.nanoTime();
        int num = 30000;
        BigInteger factorial = BigInteger.valueOf(1);
        for (int i = 1; i <= num; i++) {
            factorial = factorial.multiply(BigInteger.valueOf(i));
        }
        System.out.println("The factorial of " + num + " is: " + factorial);
        System.out.println(System.nanoTime() - tInit);
    }
}

File: Factorial/Factorial.rs
use std::time::SystemTime;
extern crate num;
use num::{BigInt, One};

fn factorial(n: u64) -> BigInt {
    let mut result = BigInt::one();
    for i in 2..=n {
        result = result * BigInt::from(i);
    }
    result
}
fn main() {
    let t_init = SystemTime::now();
    let num = 30000;
    let result = factorial(num);
    println!("The factorial of {} is {}", num, result);
   println!("{}", t_init.elapsed().unwrap().as_nanos());
}